name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/csv/CSVImporter.java

```diff
@@ -290,6 +293,200 @@ public M parseObject(String source, ParsePosition pos)
+        {
+            for (Security security : securityList)
+            {
+                if (security.getIsin() != null)
+                {
+                    String ISIN = security.getIsin().trim();
+                    if (CheckISIN(ISIN))
+                    {
+                        this.isinList.add(ISIN);
+                    }
+                }
+            }
+        }
+
+        @Override
+        public StringBuffer format(Object obj, StringBuffer toAppendTo, FieldPosition pos)
+        {
+            String s = (String)obj;
+            if (s == null)
+                throw new IllegalArgumentException();
+
+            return toAppendTo.append(s);
+        }
+        /**
+         * Calculates the Double-Add-Double from String src (length 11 chars)
+         * alternatively checks validity (length 12 chars, result 0 if valid else undefinied.
+         * http://www.foxtrot-uniform-charlie-kilo.eu/programming/java-eu/isin-check.html
+         * https://rosettacode.org/wiki/Validate_International_Securities_Identification_Number#Java
+         */
+        public static boolean luhnTest(String number)
+        {
+            int s1 = 0, s2 = 0;
+            String reverse = new StringBuffer(number).reverse().toString();
+            for(int i = 0 ;i < reverse.length();i++)
+            {
+                int digit = Character.digit(reverse.charAt(i), 10);
+                if(i % 2 == 0)
+                {//this is for odd digits, they are 1-indexed in the algorithm
+                    s1 += digit;
+                }
+                else
+                {//add 2 * digit for 0-4, add 2 * digit - 9 for 5-9
+                    s2 += 2 * digit;
+                    if(digit >= 5)
+                    {
+                        s2 -= 9;
+                    }
+                }
+            }
+            return (s1 + s2) % 10 == 0;
+        }
+
+        static boolean CheckISIN(String isin)
+        {
+            isin = isin.trim().toUpperCase();
+            if (!isin.matches("^"+ ISINpattern + "$"))
+                return false;
+            StringBuilder sb = new StringBuilder();
+            for (char c : isin.substring(0, 12).toCharArray())
+                sb.append(Character.digit(c, 36));
+
+            return luhnTest(sb.toString());
+        }
+         /** END **/
+
+        @Override
+        public Object parseObject(String source, ParsePosition pos)
+        {
+            int parseSuccessIndex = source.length();
+            source = source.trim();
+            if (pos == null)
+                throw new NullPointerException();
+            Object rObj = null;
+
+            Pattern pattern = Pattern.compile(" (" + ISINpattern + ") ");
+            Matcher matcher = pattern.matcher(source);
+            boolean success = false;
+            if (matcher.find())
+            {
+                source  = matcher.group(1);
+            }
+
+            if (source.equals("NA0123456789"))
```

Comment:
Warum ist hier eine ISIN direkt im Code?

Reply:
Abend Andreas,
good catch. In meinem Daten verwende ich das als "DummyISIN". Macht aber im Allgemeinen zugegeben keinen wirklichen Sinn. 
Nicolas